### PR TITLE
Add mapping from Grouparoo fields to Iterable nested objects.

### PR DESCRIPTION
**Pull request by pauloouriques**
- add dot notation to object coversion;
- add flag mergeNestedObjects.

### CODE REVIEW

File: plugins/@grouparoo/iterable/jest.config.js [hunk 1]
```diff
@@ -1,8 +1,6 @@
-const { helper } = require("@grouparoo/spec-helper");
-
 module.exports = {
   testEnvironment: "node",
-  testTimeout: helper.longTime,
+  testTimeout: 60 * 2 * 1000,
```
Comment (evantahler) [inline plugins/@grouparoo/iterable/jest.config.js]: A shame!

File: plugins/@grouparoo/iterable/src/lib/export/exportProfile.ts [hunk 2]
```diff
@@ -111,6 +116,7 @@ export const sendProfile: ExportProfilePluginMethod = async ({
       );
     }
 
+    payload["mergeNestedObjects"] = true;
```
Comment (evantahler) [inline plugins/@grouparoo/iterable/src/lib/export/exportProfile.ts:118]: This implies that in all cases, we can send incomplete updates to an Iterable object and we will merge the existing data with what we have.  Cool!

What doest it mean when we want to remove a piece of data?  For example, in your test you could set `addres.city` and then remove it... would it be removed from Itterable?

File: plugins/@grouparoo/iterable/__tests__/export/export-profile.ts [hunk 3]
```diff
@@ -182,7 +183,15 @@ describe("iterable/exportProfile", () => {
...
       oldProfileProperties: {},
-      newProfileProperties: { email, name },
+      newProfileProperties: {
+        email,
+        name,
+        defaultAddress: {
+          street: street1,
+          number: number1,
+          zipcode: zipcode,
+        },
```
Comment (evantahler) [inline plugins/@grouparoo/iterable/__tests__/export/export-profile.ts]: I don't think that it's likely Grouparoo would provide an object to the plugin like this

Comment (evantahler) [reply plugins/@grouparoo/iterable/__tests__/export/export-profile.ts]: The `.` way of defining properties seems more likely

**Review by evantahler — CHANGES_REQUESTED** [review]
Good start! 

I'd like to see more testing on what it means to remove a property from an Itterable Object, and now it relates to `mergeNestedObjects`

File: plugins/@grouparoo/iterable/__tests__/export/export-profile.ts (hunk 3 above)
Comment (pauloouriques) [reply plugins/@grouparoo/iterable/__tests__/export/export-profile.ts]: This is just to add the object there for the first time. But I agree we should use dot notation even the first time.

**Review by evantahler — APPROVED** [review]
I think this is a great start, and we should merge this in and get it out to our customers!

I'm still interested to know what it means to stop sending an object property.

If we had previously sent `address.city` and then stopped sending `address.city`, I think I read this as Iterable will keep `address.city` on the destination, and so unless we send `""`, there's no way to remove it.

**Comment by evantahler** [comment]
> I think this is a great start, and we should merge this in and get it out to our customers!
> 
> I'm still interested to know what it means to stop sending an object property.
> 
> If we had previously sent `address.city` and then stopped sending `address.city`, I think I read this as Iterable will keep `address.city` on the destination, and so unless we send `""`, there's no way to remove it.

@pauloouriques pointed out that this is already tested here https://github.com/grouparoo/grouparoo/blob/7b18dc8ead9ae28c9c549ffe720fce74d38858ea/plugins/%40grouparoo/intercom/__tests__/integration/export-profile-lead.ts#L164-L193 We are good to go!

**Pull request merged**